Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp

```diff
@@ -185,7 +186,7 @@ class UserMediaCaptureManagerProxySourceProxy final
     {
         ASSERT(isMainRunLoop());
 
-        m_pendingAction = m_pendingAction->isResolved() ? action() : m_pendingAction->whenSettled(RunLoop::main(), WTFMove(action));
+        m_pendingAction = protectedPendingAction()->isResolved() ? action() : m_pendingAction->whenSettled(RunLoop::protectedMain(), WTFMove(action));
```

Comment:
Let's' take a:
RefPtr protectedPendingAction = this->protectedPendingAction().
And use it for both isResolved and whenSettled.

Reply:
Fixed!